### DISCUSSION TITLE
GUI apps / usage of (X) Window Server

### DISCUSSION THREAD

**Discussion by Asqiir** [nix-community/nix-on-droid#281] [General]
Hi!

I'd like to install some GUI app (firefox-mobile, in my case) to my nix-on-droid setup. The installation was fine so far.

Why would I want that? Mostly, because nixpkgs is a large, maintained, official-ish repository for open-source software. More official than most of the stuff on f-droid.

But starting `firefox` just returned, that the `DISPLAY` variable was not set. I tried different X window servers.

* I was not able to connect to the [X Server app](https://f-droid.org/packages/au.com.darkside.xdemo/).
* [XServer XSDL](https://play.google.com/store/apps/details?id=x.org.server) worked ... sort of. Downsides: only on Google play store. Worked only wonkily. Forced to landscape mode. I had to navigate a cursor. Digital keyboard did not appear as needed. Only used ~3/4 of the whole screen
* I was able to set up [termux-x11](https://github.com/termux/termux-x11) – on termux. It did not connect to nix-on-droid.

So neither was satisfying. What X application do you use for GUI programs?

Personally, I think the termux-x11 thing should be easiest to integrate into nix-on-droid, as it's based on the termux app. (Right?)

**Comment by t184256** [comment]
> What X application do you use for GUI programs?

Never tried any (not interested), but there sure is [at least one discussion about it](https://github.com/t184256/nix-on-droid/issues/75).

> Personally, I think the termux-x11 thing should be easiest to integrate into nix-on-droid, as it's based on the termux app. (Right?)

No idea, I'm absolutely not an Android programmer. Is there a wayland one, maybe? Anyway, if you do figure it out, pull requests to [nix-on-droid-app](https://github.com/t184256/nix-on-droid-app) are welcome, but I'm not interested in doing the development work for it.
